src/opower/opower.py

```diff
@@ -181,85 +191,115 @@ async def _async_authorize(self, url: str) -> None:
+            async with self.session.get(url) as resp:
+                result = await resp.json()
+                if DEBUG_LOG_RESPONSE:
+                    _LOGGER.debug("Fetched: %s", json.dumps(result, indent=2))
+            for forecast in result["accountForecasts"]:
+                forecasts.append(
+                    Forecast(
+                        account=Account(
+                            customer=customer,
+                            uuid=forecast["accountUuids"][0],
+                            utility_account_id=str(
+                                forecast["preferredUtilityAccountId"]
+                            ),
+                            meter_type=MeterType(forecast["meterType"]),
+                        ),
+                        start_date=date.fromisoformat(forecast["startDate"]),
+                        end_date=date.fromisoformat(forecast["endDate"]),
+                        current_date=date.fromisoformat(forecast["currentDate"]),
+                        unit_of_measure=UnitOfMeasure(forecast["unitOfMeasure"]),
+                        usage_to_date=float(forecast.get("usageToDate", 0)),
+                        cost_to_date=float(forecast.get("costToDate", 0)),
+                        forecasted_usage=float(forecast.get("forecastedUsage", 0)),
+                        forecasted_cost=float(forecast.get("forecastedCost", 0)),
+                        typical_usage=float(forecast.get("typicalUsage", 0)),
+                        typical_cost=float(forecast.get("typicalCost", 0)),
+                    )
+                )
         return forecasts
 
-    async def _async_get_customer(self):
-        """Get information about the current customer."""
-        # Cache the customer data
-        if not self.customer:
+    async def _async_get_customers_data(self):
+        """Get information about the customers associated to the account."""
+        # Cache the customers data
+        if not self.customers_data:
             async with self.session.get(
                 "https://"
                 f"{self.utility.subdomain()}"
                 ".opower.com/ei/edge/apis/multi-account-v1/cws/"
                 f"{self.utility.subdomain()}"
-                # Alternative to get multiple accounts:
-                # /customers?offset=0&batchSize=100&addressFilter=
-                "/customers/current"
+                "/customers?offset=0&batchSize=100&addressFilter="
             ) as resp:
-                self.customer = await resp.json()
+                resp_json = await resp.json()
+                assert "customers" in resp_json
+                self.customers_data = resp_json["customers"]
                 if DEBUG_LOG_RESPONSE:
-                    _LOGGER.debug("Fetched: %s", json.dumps(self.customer, indent=2))
-        assert self.customer
-        return self.customer
+                    _LOGGER.debug(
+                        "Fetched: %s", json.dumps(self.customer_data, indent=2)

```

Comment:
self.customer_data is undefined, it should had been customers_data. Anyway it's better to debug log the json response instead